nextest-metadata/src/test_list.rs

```diff
@@ -485,7 +485,7 @@ pub struct RustBuildMetaSummary {
 
     /// The target platforms used while compiling the Rust artifacts.
     #[serde(default)]
-    pub target_platforms: Vec<PlatformSummary>,
+    pub target_platforms: Vec<BuildPlatformsSummary>,
```

Comment:
Ahhh, I see. I misunderstood what compatibility meant in this context, and I apologize for that. I think we do want to keep compatibility around in this case.

Also, this was my terrible oversight when I first put in `target_platforms`, I wish I'd have added this `BuildPlatformsSummary` struct at the time. I humbly apologize for the extra work you have to do here.

As far as handling this goes, I can think of three general directions.

1. Write out the old `target_platforms` field, and add a new field (say `targets`) which has the full context. Deprecate `target_platforms` and ask users to use the new field if it's present. (This may want to be an `Option<Vec<BuildPlatformsSummary>>`. (This is similar to `target_platform` below.)

   Hopefully, this adds enough indirection that will be the last time we'd need to do that.

2. Retain compatibility with the old fields in https://docs.rs/target-spec/3.1.0/src/target_spec/summaries.rs.html#33 by manually duplicating them, and add the `sysroot` as another field next to it.
    * One way to do this would be to use `serde(flatten)` -- but that requires internal buffering, and in general, serde has [numerous issues](https://github.com/serde-rs/serde/issues/1183) with internal buffering.
    * Another way would be to manually duplicate the fields. But this is a challenge for two reasons. One is that the struct is `#[non_exhaustive]` https://docs.rs/target-spec/3.1.0/src/target_spec/summaries.rs.html#32, and two is that new semantically-important fields could be added to target-spec in the future. (I'm the maintainer of target-spec as well, so I can make changes to it if they make sense.)
    
      The former we could probably add some kind of builder for, but the latter is uncomfortably easy to miss details for.
      
      We could have some kind of convoluted `PlatformSummaryBuilder` in `target-spec` that produces an error if not all fields were present, I guess. But I'm not convinced of the value of all that extra work.

3. Add another vector next to `target_platforms` called (say) `target_platforms_extra`, and promise that it's the same length as `target_platforms` and corresponds 1:1 to it (i.e. users can `zip` the two vectors).

I think between all of these options I like 1 the most. It is a little bit of extra work, but sets us up best for the future. (At some point in the future, we'll probably remove both `target_platform` and `target_platforms` with the appropriate warnings.)

Reply:
Adopt approach 1, but name the field as `target_platforms_v2` to make it explicit that `target_platforms` is deprecated. WDYT?